pkg/lib/registry/registry.go

```diff
@@ -357,6 +358,23 @@ func (r RegistryUpdater) DeprecateFromRegistry(request DeprecateFromRegistryRequ
 	}
 
 	deprecator := sqlite.NewSQLDeprecatorForBundles(dbLoader, toDeprecate)
+
+	// Check for deprecation of head of default channel. If deprecation request includes heads of all other channels,
+	// then remove the package entirely. Otherwise, deprecate provided bundles. This enables deprecating an entire package.
+	// By default deprecating the head of default channel is not permitted.
+	if request.AllowPackageRemoval {
```

Comment:
This is more related to package removal by itself, but what would this mean for previously deprecated entries? They persist in the deprecated table as just a bundle name when the package is removed

Reply:
That's true -- if I understand correctly this is being tracked in https://bugzilla.redhat.com/show_bug.cgi?id=1982781 which we will address separately. I didn't want to tackle that in this PR.